File: problems/1423_maximum_points_you_can_obtain_from_cards/Solution.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n = cardPoints.length;
        int windowSize = n - k;
        int window = 0;
        for (int i = 0; i < windowSize; i++) window += cardPoints[i];
        int minWindow = window;
        int total = window;
        for (int i = windowSize; i < n; i++) {
            total += cardPoints[i];
            window += cardPoints[i] - cardPoints[i - windowSize];
            minWindow = Math.min(minWindow, window);
        }
        return total - minWindow;
    }
}

File: problems/1423_maximum_points_you_can_obtain_from_cards/solution.py
from typing import List


class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)
        window = sum(cardPoints[:n - k])
        min_window = window
        for i in range(n - k, n):
            window += cardPoints[i] - cardPoints[i - (n - k)]
            min_window = min(min_window, window)
        return sum(cardPoints) - min_window
